adapters/repos/db/index.go

```diff
@@ -1131,7 +1134,9 @@ func (i *Index) objectSearchByShard(ctx context.Context, limit int, filters *fil
 					return fmt.Errorf(
 						"local shard object search %s: %w", shard.ID(), err)
 				}
-			} else {
+				// If a class is replication-enabled, a request with a consistency level
+				// of ONE should not cause DB to attempt contact any other nodes
+			} else if !i.replicationEnabled() || replProps.ConsistencyLevel != string(replica.One) {
```

Comment:
Assuming identical database state at query time, the number of returned objects should remain consistent regardless of replication status. The data is distributed across n shards. The search should span all shards regardless of their location, and then proceed to check the consistency of the result set.

Searching only local shards while replication is enabled with ONE would completely disrupt this behaviour